web/tests/narrow_unread.test.js

```diff
@@ -132,6 +132,28 @@ run_test("get_unread_ids", () => {
     unread_ids = candidate_ids();
     assert.deepEqual(unread_ids, [stream_msg.id]);
 
+    // For a search using `with` operator, our candidate ids
+    // will be the messages present in the channel/topic
+    // containing the message for which the `with` operand
+    // is id to.
+    terms = [
+        {operator: "channel", operand: sub.name},
+        {operator: "topic", operand: "my topic"},
+        {operator: "with", operand: 101},
+    ];
+    set_filter(terms);
+    unread_ids = candidate_ids();
+    assert.deepEqual(unread_ids, [101]);
+
+    terms = [
+        {operator: "channel", operand: sub.name},
+        {operator: "topic", operand: "my topic"},
+        {operator: "with", operand: 102},
+    ];
+    set_filter(terms);
+    unread_ids = candidate_ids();
+    assert.deepEqual(unread_ids, []);
+
```

Comment:
This test doesn't actually verify the core logic that we're trying to test, because the message is already in the right topic. Never write tests just to get coverage -- the whole point of having a test is to actually verify the code is correct. A good way to verify if your test is good is to check if the test passes without the actual code changes in the PR -- if it does, it's a bad tests, and while I didn't actually check, I'm pretty sure this test code passes on `main`. I rewrote it as follows:

``` diff
diff --git a/web/tests/narrow_unread.test.js b/web/tests/narrow_unread.test.js
index fb153d2c84..38d6d8fe6c 100644
--- a/web/tests/narrow_unread.test.js
+++ b/web/tests/narrow_unread.test.js
@@ -77,8 +77,19 @@ run_test("get_unread_ids", () => {
         display_recipient: [{id: alice.user_id}],
     };
 
+    const other_topic_message = {
+        id: 103,
+        type: "stream",
+        stream_id: sub.stream_id,
+        topic: "another topic",
+        unread: true,
+        mentioned: false,
+        mentioned_me_directly: false,
+    };
+
     message_store.update_message_cache(stream_msg);
     message_store.update_message_cache(private_msg);
+    message_store.update_message_cache(other_topic_message);
 
     stream_data.add_sub(sub);
 
@@ -132,28 +143,6 @@ run_test("get_unread_ids", () => {
     unread_ids = candidate_ids();
     assert.deepEqual(unread_ids, [stream_msg.id]);
 
-    // For a search using `with` operator, our candidate ids
-    // will be the messages present in the channel/topic
-    // containing the message for which the `with` operand
-    // is id to.
-    terms = [
-        {operator: "channel", operand: sub.name},
-        {operator: "topic", operand: "my topic"},
-        {operator: "with", operand: 101},
-    ];
-    set_filter(terms);
-    unread_ids = candidate_ids();
-    assert.deepEqual(unread_ids, [101]);
-
-    terms = [
-        {operator: "channel", operand: sub.name},
-        {operator: "topic", operand: "my topic"},
-        {operator: "with", operand: 102},
-    ];
-    set_filter(terms);
-    unread_ids = candidate_ids();
-    assert.deepEqual(unread_ids, []);
-
     terms = [{operator: "is", operand: "mentioned"}];
     set_filter(terms);
     unread_ids = candidate_ids();
@@ -224,6 +213,32 @@ run_test("get_unread_ids", () => {
         flavor: "cannot_compute",
     });
 
+    // For a search using `with` operator, our candidate ids
+    // will be the messages present in the channel/topic
+    // containing the message for which the `with` operand
+    // is id to.
+    //
+    // Here we use an empty topic for the operators, and show that
+    // adding the with operator causes us to see unreads in the
+    // destination topic.
+    unread.process_loaded_messages([other_topic_message]);
+    terms = [
+        {operator: "channel", operand: sub.name},
+        {operator: "topic", operand: "another topic"},
+    ];
+    set_filter(terms);
+    unread_ids = candidate_ids();
+    assert.deepEqual(unread_ids, [other_topic_message.id]);
+
+    terms = [
+        {operator: "channel", operand: sub.name},
+        {operator: "topic", operand: "another topic"},
+        {operator: "with", operand: stream_msg.id},
+    ];
+    set_filter(terms);
+    unread_ids = candidate_ids();
+    assert.deepEqual(unread_ids, [stream_msg.id]);
+
     message_lists.set_current(undefined);
     blueslip.expect("error", "unexpected call to get_first_unread_info");
     assert_unread_info({
```